### ISSUE TITLE
How to calculate the stun time of Ability which have the modifier is "modifier_stunned"

### ISSUE THREAD

**Issue by MFamilys** [skadistats/clarity#107]
Hi,Buddy! Thanks for the great replay parser.How nice work! : )
The **Clarity** is always worked  useful for me,but now here i meet a problem so i ask for help.
What i want to do is calculate the stun time for every ability of one hero in a match.
Now i am use the entity of `DOTA_COMBATLOG_MODIFIER_ADD` to get the stun time,but  i have noticed that is some ability have the modifier as "**modifier_stunned**".It was really comfused me,is there have some way to judge which ability is this modifier belong to? 
What's more,I see there have a field in the **dota_modifiers.proto** such as the ablitity ,maybe is can helpful? 
In short,i just need some thing can identify if the ability have the stun effect or can recognize which ability is the modifier belong to? Thanks for your help :D

**Comment by spheenik** [comment]
There is an annotation in clarity `@OnModifierTableEntry` that you can use to determine if modifiers are added / removed from an entity. It will return a CDOTAModifierBuffTableEntry which indeed has a property ability. If you're lucky, this is a handle to the ability entity, "pointing" to the ability. You could then query the ability with `ctx->getProcessor(Entities.class)->getByHandle(*theHandle*)`.

In turn, hero entities have the handles of their abilities. 

There is a very basic [for modifiers in clarity examples](https://github.com/skadistats/clarity-examples/blob/master/src/main/java/skadistats/clarity/examples/modifiers/Main.java)

HTH! Thanks for using clarity :)

**Comment by MFamilys** [comment]
Thanks for your time.the solution is what i need.But here i still meet some Exception as follow:

- **First**:

  When I use the method as the guide by
     `if(e.getAbility!=0)
       ctx.getProcessor(Entities.class).getByHandle((Integer)e.getAbility());`
  There is a NullPointerException like:
     `java.lang.NullPointerException
	at com.mfamilys.video.combatlog.Main.onModifierTableEntry(Main.java:59)
	at skadistats.clarity.event.AbstractInvocationPoint.invoke(AbstractInvocationPoint.java:72)
	at skadistats.clarity.event.Event.raise(Event.java:23)
	at skadistats.clarity.processor.modifiers.Modifiers.onTableEntry(Modifiers.java:18)
	at skadistats.clarity.event.AbstractInvocationPoint.invoke(AbstractInvocationPoint.java:72)
	at skadistats.clarity.event.Event.raise(Event.java:23)
	at skadistats.clarity.processor.stringtables.BaseStringTableEmitter.raise(BaseStringTableEmitter.java:74)
	at skadistats.clarity.processor.stringtables.BaseStringTableEmitter.setSingleEntry(BaseStringTableEmitter.java:69)
	at skadistats.clarity.processor.stringtables.S2StringTableEmitter.decodeEntries(S2StringTableEmitter.java:120)
	at skadistats.clarity.processor.stringtables.S2StringTableEmitter.onUpdateStringTable(S2StringTableEmitter.java:58)
	at skadistats.clarity.event.AbstractInvocationPoint.invoke(AbstractInvocationPoint.java:72)
	at skadistats.clarity.event.Event.raise(Event.java:23)
	at skadistats.clarity.processor.reader.InputSourceProcessor.processEmbedded(InputSourceProcessor.java:248)
	at skadistats.clarity.event.AbstractInvocationPoint.invoke(AbstractInvocationPoint.java:72)
	at skadistats.clarity.event.Event.raise(Event.java:23)
	at skadistats.clarity.processor.reader.InputSourceProcessor.processSource(InputSourceProcessor.java:170)
	at skadistats.clarity.event.AbstractInvocationPoint.invoke(AbstractInvocationPoint.java:72)
	at skadistats.clarity.event.Event.raise(Event.java:23)
	at skadistats.clarity.processor.runner.AbstractRunner.runWith(AbstractRunner.java:78)
	at com.mfamilys.video.combatlog.Main.run(Main.java:179)
	at com.mfamilys.video.combatlog.Main.main(Main.java:185)`


-  **Second**:

     When i get the Modifier Name from "ModifierNames" StringTable by e.getIndex().That is seem some wrong.
     The ModifierName is not matching the Match such as appear the modifier with the hero name not exist in Match.Maybe the way i use is wrong ?

BTW,Thanks your code and patience for my problem. : )

**Comment by spheenik** [comment]
What is e? A `CDOTAModifierBuffTableEntry`, I suppose? I don't know exactly, but if it's not index, can you try `modifier_class`?

**Comment by MFamilys** [comment]
Yes.e is a **CDOTAModifierBuffTableEntry**.
BTW,The `modifier_class` is exactly works, thanks a lot for the answer! :+1:

**Comment by STRATZ-Ken** [comment]
@spheenik Can we confirm if this StringTable still exists?  I am trying to convert the `modifier_class` to the modifier name but cannot do so as I dont see a StringTable named ModifierNames anymore.  Any ideas?
